LibreNMS/Alert/Transport/Pagerduty.php

```diff
@@ -43,7 +43,13 @@ public function deliverAlert($obj, $opts)
             $protocol['event_type'] = 'trigger';
         }
         foreach ($obj['faults'] as $fault => $data) {
-            $protocol['details'][] = $data['string'];
+            $rough_details = array_filter(explode('; ', $data['string']));
```

Comment:
I think there is an existing array version stored in the faults array so you don't have to parse the string.  Did you check that?

Reply:
I didn't see an equivalent 'array' nested element that could directly be copied.  Looking over it again though I think there is a slightly easier way to do this.  I can test and resubmit tomorrow.

```
$details = array()
foreach ($obj['faults'] as $fault => $data) {
    $details = $data;
    unset($details['string']);
}
$protocol['details'] = $details;
```

Reference lines around 497, 524
https://github.com/librenms/librenms/blob/master/includes/alerts.inc.php

Reply:
That's exactly what I was thinking sorry if I explained it poorly.  I wasn't sure of the format.